src/main/java/frc/robot/subsystems/led/Led2023.java

```diff
@@ -664,64 +746,45 @@ public void setAlternateColorsUp(COLORS_467 colorOne, COLORS_467 colorTwo, Color
     // t=1, 2, or 3. sets top 1/3, mid 1/3, or lower 1/3
-    double start;
-    double end;
-    int othersLow = 190;
-    int othersHi = 190;
-    int othLow = 190;
-    int othHi = 190;
-    boolean valid = true;
+    int start;
+    int end;
+
     if (preSet == 1) {
-      start = 0;
-      end = Math.ceil(RobotConstants.get().led2023LedCount() / 3);
-      othersLow = (int) Math.ceil(RobotConstants.get().led2023LedCount() / 3) + 1;
-      othersHi = (int) Math.ceil(RobotConstants.get().led2023LedCount()) - 1;
+      start = (int) topStart;
+      end = (int) topEnd;
     } else if (preSet == 2) {
-      start = Math.floor(RobotConstants.get().led2023LedCount() / 3);
-      end =
-          Math.ceil(
-              RobotConstants.get().led2023LedCount()
-                  - (RobotConstants.get().led2023LedCount() / 3));
-      othersLow = 0;
-      othersHi = (int) Math.floor(RobotConstants.get().led2023LedCount() / 3) - 1;
-      othLow =
-          (int)
-                  Math.ceil(
-                      RobotConstants.get().led2023LedCount()
-                          - (RobotConstants.get().led2023LedCount() / 3))
-              + 1;
-      othHi = RobotConstants.get().led2023LedCount() - 1;
-
-    } else if (preSet == 3) {
-      start =
-          Math.floor(
-              RobotConstants.get().led2023LedCount()
-                  - (RobotConstants.get().led2023LedCount() / 3));
-      end = Math.ceil(RobotConstants.get().led2023LedCount());
-      othersLow =
-          (int)
-                  Math.floor(
-                      RobotConstants.get().led2023LedCount()
-                          - (RobotConstants.get().led2023LedCount() / 3))
-              + 1;
-      othersHi = RobotConstants.get().led2023LedCount() - 1;
+      start = (int) midStart;
+      end = (int) midEnd;
+
     } else {
-      start = 0;
-      end = RobotConstants.get().led2023LedCount();
+      start = (int) bottomStart;
+      end = (int) bottomEnd;
     }
-    for (int i = (int) start; i < end; i++) {
+    for (int i = start; i < end; i++) {
       ledStrip.setLED(i, color.getColor());
     }
-    if (othersHi == 190 || othHi == 190 || othersHi == 190 || othersLow == 190) {
-      valid = false;
-    } else if (valid == true) {
-      for (int i = othersLow; i <= othersHi; i++) {
-        ledStrip.setLED(i, COLORS_467.Black.getColor());
+
+    ledStrip.update();
+  }
+
+  public void victoryLeds(COLORS_467 topColor, COLORS_467 bottomColor, COLORS_467 bgColor) {
+    if (purpleTimer.hasElapsed(0.9)) {
+      purpleTimer.reset();
+    } else if (purpleTimer.hasElapsed(0.4)) {
+      for (int i = 0; i < RobotConstants.get().led2023LedCount(); i++) {
+        for (int brightness = 0; brightness <= 256; brightness++) {
+      ledStrip.setRGB(i, topColor.red * brightness, topColor.blue * brightness, topColor.green * brightness);
+      ledStrip.setRGB(i, bottomColor.red * brightness, bottomColor.blue * brightness, bottomColor.green * brightness);
```

Comment:
It seems like you're setting the same LED index to two different colors, one right after the other.  I think the first one isn't actually doing anything since the second one overwrites it.